bin/import_gram_vaani.py

```diff
@@ -0,0 +1,302 @@
+        _logger.info("Parsing csv file...%s", os.path.abspath(csv_filename))
+        data = pd.read_csv(
+            os.path.abspath(csv_filename),
+            names=["piece_id","audio_url","transcript_labelled","transcript","labels","content_filename","audio_length","user_id"],
+            usecols=["audio_url","transcript","audio_length"],
+            skiprows=[0],
+            engine="python",
+            encoding="utf-8",
+            quotechar='"',
+            quoting=csv.QUOTE_ALL,
+        )
+        data.dropna(inplace=True)
+        _logger.info("Parsed %d lines csv file." % len(data))
+        return data
+
+class GramVaaniDownloader:
+    """GramVaaniDownloader downloads a GramVaani dataset.
+    Args:
+      gram_vaani_csv (GramVaaniCSV): A GramVaaniCSV representing the data to download
+      directory (str): The path to download the data from
+    Attributes:
+        data (:class:`pandas.DataFrame`): `pandas.DataFrame` Containing the GramVaani csv data
+    """
+
+    def __init__(self, gram_vaani_csv, directory):
+        self.directory = directory
+        self.data = gram_vaani_csv.data
+
+    def download(self):
+        """Downloads the data associated with this instance
+        Return:
+          mp3_directory (os.path): The directory into which the associated mp3's were downloaded
+        """
+        mp3_directory = self._pre_download()
+        self.data.swifter.apply(func=lambda arg: self._download(*arg, mp3_directory), axis=1, raw=True)
+        return mp3_directory
+
+    def _pre_download(self):
+        mp3_directory = path.join(self.directory, "mp3")
+        if not path.exists(self.directory):
+            _logger.info("Creating directory...%s", self.directory)
+            os.mkdir(self.directory)
+        if not path.exists(mp3_directory):
+            _logger.info("Creating directory...%s", mp3_directory)
+            os.mkdir(mp3_directory)
+        return mp3_directory
+
+    def _download(self, audio_url, transcript, audio_length, mp3_directory):
+        if audio_url == "audio_url":
+            return
+        mp3_filename = path.join(mp3_directory, os.path.basename(audio_url))
+        if not path.exists(mp3_filename):
+            _logger.debug("Downloading mp3 file...%s", audio_url)
+            urllib.request.urlretrieve(audio_url, mp3_filename)
+        else:
+            _logger.debug("Already downloaded mp3 file...%s", audio_url)
+
+class GramVaaniConverter:
+    """GramVaaniConverter converts the mp3's to wav's for a GramVaani dataset.
+    Args:
+      directory (str): The path to download the data from
+      mp3_directory (os.path): The path containing the GramVaani mp3's
+    Attributes:
+        directory (str): The target directory passed as a command line argument
+        mp3_directory (os.path): The path containing the GramVaani mp3's
+    """
+
+    def __init__(self, directory, mp3_directory):
+        self.directory = directory
+        self.mp3_directory = Path(mp3_directory)
```

Comment:
nit: Why not using `Path` class also for the other paths?

Reply:
Looking at the code in retrospect, I think this Path is there as the code...

```python
...
for mp3_filename in self.mp3_directory.glob('**/*.mp3'):
    wav_filename = path.join(wav_directory, os.path.splitext(os.path.basename(mp3_filename))[0] + ".wav")
    if not path.exists(wav_filename):
...
```

would have to change to

```python
...
mp3_directory = Path(self.mp3_directory)
for mp3_filename in mp3_directory.glob('**/*.mp3'):
    wav_filename = path.join(wav_directory, os.path.splitext(os.path.basename(mp3_filename))[0] + ".wav")
    if not path.exists(wav_filename):
...
```

If I was consistent in using strings.

If I used Path everywhere I don't think the usage of self.directory would change.

I'm not sure if there's a big "win" for switching all code to use Path.